src/query/storages/fuse/src/operations/read/parquet_data_source_deserializer.rs

```diff
@@ -110,8 +126,138 @@ impl DeserializeDataTransform {
+                        .build_keys_state(&[(probe_column, data_type)], data_block.num_rows())?;
+                    match key_state {
+                        KeysState::Column(Column::Number(NumberColumn::UInt32(c))) => {
+                            c.iter().for_each(|key| {
+                                let hash = key.fast_hash();
+                                if filter.contains(&hash) {
+                                    bitmap.set(idx, true);
+                                }
+                                idx += 1;
+                            })
+                        }
+                        _ => unreachable!(),
+                    }
+                }
+                HashMethodKind::KeysU64(hash_method) => {
+                    let key_state = hash_method
+                        .build_keys_state(&[(probe_column, data_type)], data_block.num_rows())?;
+                    match key_state {
+                        KeysState::Column(Column::Number(NumberColumn::UInt64(c))) => {
+                            c.iter().for_each(|key| {
+                                let hash = key.fast_hash();
+                                if filter.contains(&hash) {
+                                    bitmap.set(idx, true);
+                                }
+                                idx += 1;
+                            })
+                        }
+                        _ => unreachable!(),
+                    }
+                }
+                HashMethodKind::KeysU128(hash_method) => {
+                    let key_state = hash_method
+                        .build_keys_state(&[(probe_column, data_type)], data_block.num_rows())?;
+                    match key_state {
+                        KeysState::U128(c) => c.iter().for_each(|key| {
+                            let hash = key.fast_hash();
+                            if filter.contains(&hash) {
+                                bitmap.set(idx, true);
+                            }
+                            idx += 1;
+                        }),
+                        _ => unreachable!(),
+                    }
+                }
+                HashMethodKind::KeysU256(hash_method) => {
+                    let key_state = hash_method
+                        .build_keys_state(&[(probe_column, data_type)], data_block.num_rows())?;
+                    match key_state {
+                        KeysState::U256(c) => c.iter().for_each(|key| {
+                            let hash = key.fast_hash();
+                            if filter.contains(&hash) {
+                                bitmap.set(idx, true);
+                            }
+                            idx += 1;
+                        }),
+                        _ => unreachable!(),
+                    }
+                }
+                _ => unreachable!(),
+            }
+        }
+        data_block.filter_with_bitmap(&bitmap.into())
+    }
```

Comment:
The native data source and here have many duplicate codes. Maybe we can reduce that, but it doesn't matter.

Reply:
done